src/main/java/io/confluent/avro/random/generator/Generator.java

```diff
@@ -1008,11 +1033,51 @@ private GenericEnumSymbol generateEnumSymbol(Schema schema) {
   }
 
   private GenericFixed generateFixed(Schema schema) {
-    byte[] bytes = new byte[schema.getFixedSize()];
-    random.nextBytes(bytes);
+    LogicalTypes.Decimal decimalLogicalType = getDecimalLogicalType(schema);
+    byte[] bytes;
+    if (decimalLogicalType != null) {
+      bytes = generateDecimal(decimalLogicalType);
+    } else {
+      bytes = new byte[schema.getFixedSize()];
+      random.nextBytes(bytes);
+    }
     return new GenericData.Fixed(schema, bytes);
   }
 
+  /*
+    The decimal logical type represents an arbitrary-precision signed decimal number of the form
+  unscaled × 10-scale.
+
+    A decimal logical type annotates Avro bytes or fixed types. The byte array must contain the
+  two's-complement representation of the unscaled integer value in big-endian byte order. The scale
+  is fixed, and is specified using an attribute.
+  
+    We generate a random decimal here by starting with a value of zero, then repeatedly multiplying
+  by 10^18 (one less than the maximum number of digits in a long), and adding a new random value in
+  the range [0, 10^18) generated using the Random object for this generator. This is done until the
+  precision of the current value is equal to or greater than the precision of the logical type. At
+  this point, any extra digits (of there should be at most 17) are rounded off from the value, a
+  sign is randomly selected, it is converted to big-endian two's-complement representation, and
+  returned.
+   */
+  private byte[] generateDecimal(LogicalTypes.Decimal decimalLogicalType) {
+    BigInteger bigInteger = BigInteger.ZERO;
+    final long maxIncrementExclusive = 1000000000000000000L;
+    int precision;
+    for (precision = 0; precision < decimalLogicalType.getPrecision(); precision += 18) {
+      bigInteger = bigInteger.multiply(BigInteger.valueOf(maxIncrementExclusive));
+      long increment = (long) (random.nextDouble() * maxIncrementExclusive);
```

Comment:
A double only has [15 decimal digits of precision](https://en.wikipedia.org/wiki/Double-precision_floating-point_format), so this won't be able to generate all of the digits you think it might be able to.

I would suggest either not using double, or changing your max increment to `10^15`
```suggestion
      long increment = Math.abs(random.nextLong()) % maxIncrementExclusive;
```
This fix has a distribution problem, where the wraparound makes the bottom of the range twice as likely.
You could fix this by instead using whole longs and shifting by 63, but that makes counting the digits messy.

Reply:
Nice catch. I'd like to change the max increment to just be 10^15 in order to avoid the distribution problem, but if you can find a clean solution involving longs that distributes evenly I'd happily choose that instead.